tests/manager_arg_passer.c

```diff
@@ -72,36 +23,40 @@ prep_filename(char* filename)
     return filename;
 }
 
-void
-args_clean(void** comp_args, size_t arg_count)
-{
-    for (size_t i = 0; i < arg_count; ++i)
-    {
-        free(comp_args[i]);
-    }
-    free(comp_args);
-}
-
 int
 main(int argc, char** argv)
 {
     // Initial setup
     manager_ddc = cheri_ddc_get();
     setup_intercepts();
 
-    assert(argc >= 3 && "Expect at least two arguments: binary file for compartment, and at least one argument to pass to compartment.");
+    assert(argc >= 4 && "Expect at least three arguments: binary file for compartment, entry function for compartment, and at least one argument to pass to compartment function.");
     char* file = prep_filename(argv[1]);
+    char* file_config = malloc(sizeof(file) + sizeof(comp_config_suffix) + 1);
+    strcpy(file_config, file);
+    strcat(file_config, comp_config_suffix);
+    FILE* comp_config_fd = fopen(file_config, "r");
+    free(file_config);
+    struct ConfigEntryPoint* cep = NULL;
+    size_t entry_point_count = 0;
+    char** entry_points = NULL;
+    if (comp_config_fd)
```

Comment:
What does it mean for there not to be a file descriptor for the config file?

Reply:
Ah, I had this as I initially wanted to have `main` as a default entry point, if a configuration file is not provided. I see I have not added that functionality in the current PR (and it would be hard to do, due to the current pointer argument limitation). I can leave it as a `TODO` for next PR, or quickly hardcode something to set the default.

Reply:
Either works for me.

Reply:
I've added a TODO for now; I need to investigate a bit further (as well as checking why some tests pass when they should fail, which might be related to this...).

Reply:
Is that https://github.com/capablevms/CHERI-ELF-comp/pull/7/commits/2fd7141545daad251ce44e85ca324b4907674817 ?

Reply:
Yes.

Reply:
I'm not sure I get how https://github.com/capablevms/CHERI-ELF-comp/commit/52abf26c968d54b0fc9b86d3eb56ad392026062f relates to this thread?

Reply:
The original question about there not being a config file implies the supposed need of an `else`. Since I added a sort of default entry point in that commit [1], I could add the `else` branch here to make use of that. It's just that it's a bit of a quick and dirty default option, needed to get some tests passing (unrelated to argument passing), that could be improved with more thought.

[1] https://github.com/capablevms/CHERI-ELF-comp/pull/7/commits/52abf26c968d54b0fc9b86d3eb56ad392026062f#diff-adfde91070db01742cb89ede1d5f00ff6ca15a8f01b6606c1bae07ee95b132ddR351

Reply:
OK.